libcodechecker/analyze/analysis_manager.py

```diff
@@ -224,246 +493,122 @@ def check(check_data):
-                    LOG.error(rh.analyzer_stderr)
-                return_codes = rh.analyzer_returncode
 
-                # Remove files that successfully analyzed earlier on.
-                plist_file = result_base + ".plist"
-                if os.path.exists(plist_file):
-                    os.remove(plist_file)
+                handle_failure(source_analyzer, rh, action, zip_file,
+                               result_base)
+
+                if ctu_active:
+                    LOG.error("Analyzing '" + source_file_name + "' with " +
+                              action.analyzer_type + " CTU failed.")
+                else:
+                    LOG.error("Analyzing '" + source_file_name + "' with " +
+                              action.analyzer_type + " failed.")
+
+                handle_failure(source_analyzer, rh, action, zip_file,
+                               result_base)
+
+                if ctu_active and ctu_reanalyze_on_failure:
+                    LOG.error("Try to reanalyze without CTU")
+                    # Try to reanalyze with CTU disabled.
+                    source_analyzer, analyzer_cmd, rh = \
+                        prepare_check(source, action,
+                                      analyzer_config_map,
+                                      output_dir,
+                                      context.severity_map,
+                                      skip_handler,
+                                      True)
+
+                    # Fills up the result handler with
+                    # the analyzer information.
+                    source_analyzer.analyze(analyzer_cmd,
+                                            rh,
+                                            analyzer_environment)
+
+                    if rh.analyzer_returncode == 0:
+                        handle_success(rh, result_file, result_base,
+                                       skip_handler, capture_analysis_output,
+                                       success_dir, zipfile)
+
+                        msg = "[{0}/{1}] {2} analyzed {3} without" \
+                            " CTU successfully.".format(
+                                      progress_checked_num.value,
+                                      progress_actions.value,
+                                      action.analyzer_type,
+                                      source_file_name)
+
+                        LOG.info(msg)
+                    else:
+                        LOG.error("Analyzing '" + source_file_name +
+                                  "' with " + action.analyzer_type +
+                                  " without CTU failed.")
+
+                        zip_file = result_base + '.zip'
+                        zip_file = os.path.join(failed_dir, zip_file)
+                        handle_failure(source_analyzer, rh, action,
+                                       zip_file, result_base)
+
+            if not quiet_output_on_stdout:
```

Comment:
You created two if condition with the same condition: `if not quiet_output_on_stdout:`
This can be written like so:
```python
            if not quiet_output_on_stdout:
                if rh.analyzer_returncode:
                    LOG.error('\n' + rh.analyzer_stdout)
                    LOG.error('\n' + rh.analyzer_stderr)
                else:
                    LOG.debug_analyzer('\n' + rh.analyzer_stdout)
                    LOG.debug_analyzer('\n' + rh.analyzer_stderr)
```